app/models/assessment.rb

```diff
@@ -300,6 +300,36 @@ def dump_yaml
     File.open(asmt_yaml_path, "w") { |f| f.write(YAML.dump(sort_hash(serialize))) }
   end
 
+  # If the name field in a yaml for an assessment differs from the folder name
+  # then the name field for the assessment gets overridden, which then causes
+  # the yaml file to no longer be found, and then for cleanup to fail because
+  # the folder can no longer also be found. For example a folder with
+  # homework01 and a homework01.yml file but with a name field of homework02
+  # would cause assessment.name = homework02 so then it would try to find
+  # homework02/homework02.yml. Instead we rename the name field to match the folder
+  # name.
+  # Similarly for a bunch of other fields in the yaml, if they are not provided,
+  # then we may run into issues loading the assessment
+  def validate_yaml(yaml_map)
+    unless yaml_map["general"]
+      raise "General section missing in yaml"
+    end
+
+    if yaml_map["general"]["name"] != name
+      yaml_map["general"]["name"] = name
```

Comment:
Yml file does not auto propagate the name field when it does not exist or it is different from name. Seems to be a bug that occurs for every yml file.

Reply:
But does not affect the functionality otherwise and okay to ignore if theres no quick fix.

Reply:
Going to add to another issue